aws/logs_monitoring/trace_forwarder/cmd/trace/main.go

```diff
@@ -140,22 +140,12 @@ func aggregateTracePayloadsByEnv(tracePayloads []*pb.TracePayload) []*pb.TracePa
 }
 
 func sendTracesToIntake(tracePayloads []*pb.TracePayload) error {
-	hadErr := false
 	for _, tracePayload := range tracePayloads {
 		err := edgeConnection.SendTraces(context.Background(), tracePayload, 3)
 		if err != nil {
 			fmt.Printf("Failed to send traces with error %v\n", err)
-			hadErr = true
+			return errors.New("Failed to send traces to intake")
```

Comment:
To we want to fail before attempting to finish the whole batch?

Reply:
That's a good call, can't remember why I made that change. Let me fix this.